backend/routes/admin.py

```
@@ -0,0 +1,656 @@
+async def health_check():
+    """Health check endpoint (no auth required)."""
+    try:
+        # Test database connection
+        with admin_db_manager.get_connection() as conn:
+            cursor = conn.cursor()
+            cursor.execute("SELECT 1")
+            cursor.fetchone()
+
+        return {"status": "healthy", "timestamp": datetime.now().isoformat()}
+    except Exception as e:
+        raise HTTPException(status_code=503, detail=f"Database connection failed: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)